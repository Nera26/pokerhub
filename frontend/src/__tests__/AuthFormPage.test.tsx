import { render, screen } from '@testing-library/react';
import LoginPage from '@/features/login';
import ForgotPasswordPage from '@/features/forgot-password';

jest.mock('@/components/auth/login-form', () => () => <div>Login Form</div>);
jest.mock('@/components/auth/forgot-password-form', () => () => (
  <div>Forgot Password Form</div>
));
jest.mock('@/components/auth/social-login-buttons', () => () => (
  <div>Social Login Buttons</div>
));

describe('AuthSimplePage', () => {
  it('renders login title, form, and footer link', () => {
    render(<LoginPage />);

    expect(screen.getByRole('heading', { name: /login/i })).toBeInTheDocument();
    expect(screen.getByText('Login Form')).toBeInTheDocument();
    expect(
      screen.getByRole('link', { name: /forgot password\?/i }),
    ).toHaveAttribute('href', '/forgot-password');
  });

  it('renders forgot password title, form, and footer link', () => {
    render(<ForgotPasswordPage />);

    expect(
      screen.getByRole('heading', { name: /forgot password/i }),
    ).toBeInTheDocument();
    expect(screen.getByText('Forgot Password Form')).toBeInTheDocument();
    expect(
      screen.getByRole('link', { name: /back to login/i }),
    ).toHaveAttribute('href', '/login');
  });
});
