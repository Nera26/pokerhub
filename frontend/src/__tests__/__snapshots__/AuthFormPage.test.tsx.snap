// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthFormPage renders forgot password page correctly 1`] = `
<div>
  <main
    class="flex min-h-screen flex-col items-center justify-center gap-6"
    id="main-content"
  >
    <h1
      class="text-2xl font-bold"
    >
      Forgot Password
    </h1>
    <div>
      Forgot Password Form
    </div>
    <div>
      Social Login Buttons
    </div>
    <a
      class="text-accent-yellow hover:underline"
      href="/login"
    >
      Back to Login
    </a>
  </main>
</div>
`;

exports[`AuthFormPage renders login page correctly 1`] = `
<div>
  <main
    class="flex min-h-screen flex-col items-center justify-center gap-6"
    id="main-content"
  >
    <h1
      class="text-2xl font-bold"
    >
      Login
    </h1>
    <div>
      Login Form
    </div>
    <div>
      Social Login Buttons
    </div>
    <a
      class="text-accent-yellow hover:underline"
      href="/forgot-password"
    >
      Forgot password?
    </a>
  </main>
</div>
`;

